apps/client/src/common/stores/connectionName.ts

```diff
@@ -0,0 +1,26 @@
+import { useStore } from 'zustand';
+import { createStore } from 'zustand/vanilla';
+
+interface SocketClientNameState {
+  name?: string;
+  setName: (newValue: string) => void;
+  persistName: (newValue: string) => void;
+}
+
+const clientNameKey = 'ontime-client-name';
+
+async function persistKeyToStorage(newValue: string) {
+  localStorage.setItem(clientNameKey, newValue);
+}
+
+export const socketClientName = createStore<SocketClientNameState>((set) => ({
+  name: localStorage.getItem(clientNameKey) ?? undefined,
```

Comment:
Curious to know the added benefit of `?? undefined`. Are you against `localStorage.getItem` returning `null`?

Reply:
yes, the ?? undefined is prevent null or empty string, just to narrow down the possible returns, ie: has a name or is undefined
What do you think?
BTW, really enjoy your code reviews, good stuff

Reply:
So is there something wrong with `null` that you prefer `undefined`? More like a question about preferences than logic.

Good reviews keep everyone on their feet, lol. IMO, a defining difference between a good team vs an excellent team.

Reply:
> ?? undefined is prevent null or empty string

FYI, nullish coalescing [doesn't prevent empty strings](https://stackoverflow.com/a/63924076)

Reply:
Couldn't agree more.

I generally find undefined to be more semantic as to "the value does not exist" while null would be more inclined to convey "this value is set as nothing".
Maybe that is more to do with my C++ background

Reply:
I see. So then follow your heart!